department-of-veterans-affairs/caseflow
# Justin/APPEALS-42113_CorrespondenceReviewPackageController

Resolves [Correspondence Controller needs to be Split Apart: Review Package](https://jira.devops.va.gov/browse/APPEALS-42113)

# Description
Moved review package methods from the CorrespondenceController to a new child CorrespondenceReviewPackageController. Created a spec file and moved the tests from the overall correspondence spec to the new review package spec. Made sure test verified the new routing in this spec file.

## Acceptance Criteria
- [ ] Move Correspondence Review Package code into its own child controller
- [ ] Move Correspondence Review Package spec tests into its own spec file

## Testing Plan

1. Log in as Jolly Postman and go to your queue
2. Switch views to the correspondence cases
3. Open the RailsPanel extension on Chrome
4. Find an appeal that routes to "Review Package" (the link preview should show on the bottom lefthand corner of the screen)
5. The rails panel should show the review package methods being hit on the new review package controller. The following methods should be seen in this step:
- CorrespondenceReviewPackageController#review_package
- CorrespondenceReviewPackageController#show
- CorrespondenceReviewPackageController#package_documents
- CorrespondenceReviewPackageController#pdf
(Note: some of these will run multiple times. That's normal!)
6. Next, choose "edit" next to the CMP information and change the package document type. Click Save. The following method should now be seen:
- CorrespondenceReviewPackageController#update_cmp
7. Next, under "General Information", click on a correspondence type and select a new type from the list. Click on "Save changes". The following method should now be seen:
- CorrespondenceReviewPackageController#update

Note: Previously, all of these routes would be hitting: CorrespondenceController#[method_name]. This ticket was to create the new files so that the CorrespondenceReviewPackageController#[method_name] route would be hit instead.
